lib/model/store.dart

```diff
@@ -144,6 +150,16 @@ class PerAccountStore extends ChangeNotifier {
       for (final view in _messageListViews) {
         view.maybeAddMessage(event.message);
       }
+    } else if (event is RealmUserEvent) {
+      debugPrint("server event: realm_user");
+      if (event.person.user_id == user_id) {
+        String? new_full_name = event.person.full_name;
+        if (new_full_name != null) {
+          full_name = new_full_name;
```

Comment:
I did consider both approaches. I decided on the former based on not repeating the dotted event access (ie. DRY). I'm surprised the `!` would be necessary given the null check in each case, but that seems a Dart design.

Reply:
Cool. Yeah, the reason the `!` is needed is that in Dart, every access of the form `thing.property` is in principle an access through a getter method (and similarly `thing.property = …` is through a setter method) — if the getter method happens to be implemented in a trivial way by a field on the object, that's an implementation detail of the object.

(The compiler will certainly look through that abstraction to know the difference when it's generating code, in order to make it efficient, but the language semantics maintain the abstraction.)

So from that perspective, there's no guarantee that just because `event.person.full_name` returned something non-null once, it'll return something non-null the next time.